crates/sui-indexer-alt-graphql/src/api/types/available_range.rs

```diff
@@ -78,256 +80,267 @@ impl AvailableRange {
+                            $($(
+                                filters.insert($f_to_add.to_string());
+                            )?)?
+                            let delegate_field = delegate!(_field, $delegate_field);
+                            collect_pipelines(stringify!($delegate_type), delegate_field, filters, pipelines);
+                        )?
+                        $(
+                            let $filt = filters;
+                            let $pipes = pipelines;
+                            $block
+                        )?
+                    }
+                )*
+                _ => {}
+            }
         }
 
-        // IObject fields
-        ("IObject", Some("receivedTransactions"), mut filters) => {
-            filters.insert("affectedAddress".to_string());
-            collect_pipelines("Query", Some("transactions"), filters, pipelines);
-        }
-        ("IObject", Some("objectAt" | "objectVersionsAfter" | "objectVersionsBefore"), _) => {
-            pipelines.insert("obj_versions".to_string());
-        }
-        // Object fields
-        (
-            "Object",
-            field @ Some("balance" | "balances" | "multiGetBalances" | "objects"),
-            filters,
-        ) => collect_pipelines("IAddressable", field, filters, pipelines),
-        ("Object", Some("defaultSuinsName"), filters) => {
-            collect_pipelines("IAddressable", Some("defaultSuinsName"), filters, pipelines);
-        }
-        (
-            "Object",
-            field @ Some(
-                "asMoveObject"
-                | "asMovePackage"
-                | "dynamicField"
-                | "dynamicFields"
-                | "dynamicObjectField"
-                | "multiGetDynamicFields"
-                | "multiGetDynamicObjectFields",
-            ),
-            filters,
-        ) => collect_pipelines("IMoveObject", field, filters, pipelines),
-        (
-            "Object",
-            field @ Some(
-                "digest"
-                | "objectAt"
-                | "objectBcs"
-                | "objectVersionsAfter"
-                | "objectVersionsBefore"
-                | "owner"
-                | "previousTransaction"
-                | "storageRebate"
-                | "version",
-            ),
-            filters,
-        ) => collect_pipelines("IObject", field, filters, pipelines),
+        /// Map from (Type, Field) to Option<(DelegateType, DelegateField)> for testing.
+        /// Maps to None if the field is not delegated. The "*" wildcard is expanded.
+        #[cfg(test)]
+        static TYPE_FIELD_DELEGATIONS: std::sync::LazyLock<std::collections::HashMap<(&'static str, &'static str), (&'static str, &'static str)>> =
+            std::sync::LazyLock::new(|| {
+                let mut map = std::collections::HashMap::new();
+                $(
+                    let type_name = stringify!($type);
+                    let fields = [$(stringify!($field),)*];
```

Comment:
I wonder if you can just triply-nest the `$(...)`-s so that you can generate a flat list of tuples like I suggested above?

```rust
$($($((
    stringify!($type),
    stringify!($field),
    stringify!($delegate_type),
    delegate!($field, $delegate_field), 
))?),*),*
```

The outer one is for `$type`, then for `$field`, then for `$delegate_type`.